code/game/atoms.dm

```diff
@@ -194,6 +199,14 @@ var/global/list/ghdel_profiling = list()
 /atom/proc/Bumped(AM as mob|obj)
 	return
 
+/atom/proc/setDensity(var/density)
```

Comment:
Suggestion: make this do nothing if the new density is the same as the current density.